owners/src/owners_check.js

```diff
@@ -83,20 +84,52 @@ class OwnersCheck {
     this.tree = await this.parser.parseOwnersTree();
     this.approvers = await this._getApprovers();
     this.changedFiles = await this.github.listFiles(this.pr.number);
-    this.fileTreeMap = this.tree.buildFileTreeMap(this.changedFiles);
     this.initialized = true;
   }
 
+  /**
+   * Runs the owners check and, if necessary, the reviewer selection algorithm.
+   *
+   * @return {CheckRun} a GitHub check-run with approval and reviewer info.
+   */
+  async run() {
+    if (!this.initialized) {
+      await this.init();
+    }
+
+    const fileTreeMap = this.tree.buildFileTreeMap(this.changedFiles);
+    const coverageText = this.buildCurrentCoverageText(fileTreeMap);
+
+    // Note: This starts removing files from the fileTreeMap that are already
+    // approved, so we build the coverage text first.
+    this.changedFiles.forEach(filename => {
+      const subtree = fileTreeMap[filename];
+      if (this._hasOwnersApproval(filename, subtree)) {
+        delete fileTreeMap[filename];
+      }
+    });
+    const passing = !Object.keys(fileTreeMap).length;
+    const summary = `The check was a ${passing ? 'success' : 'failure'}!`;
```

Comment:
Not a huge fan of this technical summary text. Maybe something like `This PR has/doesn't have all required ownership approvals`?

Reply:
The summary (and main output) support markdown, and I plan to enhance the text for all cases. Notably, failing checks should include the reviewer suggestions in the summary, rather than requiring a click-through (at least until reviewers are auto-assigned). Since that's not the objective of this PR though, I'm leaving this as is until I work out how all cases will/should look once the bot is enabled.

Reply:
👍